### PR TITLE
Feature request : add cvar to disable machine gun recoil in single player #741 (baseq2)

### PR DESCRIPTION
**Pull request by De-Seppe**
This adds the cvar machinegun_norecoil
It should only impact non-deathmatch games.
I used the aimfix cvar as a template to make this change.

The cvar will only work if the game.dll supports it.

### CODE REVIEW

File: doc/040_cvarlist.md [hunk 1]
```diff
@@ -48,6 +48,12 @@ it's `+set busywait 0` (setting the `busywait` cvar) and `-portable`
 * **aimfix**: Fix aiming. When set to to `0` (the default) aiming is
   slightly inaccurate, bullets and the like have a little drift. When
   set to `1` they hit exactly were the crosshair is.
+  
+* **machinegun_norecoil**: Disable machine gun recoil in single player. 
+  By default this is set to `0`, this keeps the original machine gun 
+  recoil in single player. When set to `1` the recoil is disabled in
+  single player, the same way as in multiplayer.
+  This cvar only works if the game.dll implements this behaviour.
```
Comment (Yamagi) [inline doc/040_cvarlist.md:56]: Can we rename the cvar to `g_machinegun_norecoil`? I know, some of the older cvars are missnamed, but we should do it right for new ones.

File: src/game/player/weapon.c [hunk 2]
```diff
@@ -1258,7 +1258,8 @@ Machinegun_Fire(edict_t *ent)
 			ent->client->machinegun_shots = 9;
 		}
 	}
-
+	
+
```
Comment (Yamagi) [inline src/game/player/weapon.c:1262]: Here's one superfluous newline.

**Review by Yamagi — CHANGES_REQUESTED** [review]
I've added two small comments inline. Besides that it's fine and good work. :)

One small hint: Don't create pull requests from your master / main branch. All commits pushed to the branch the pull requested was created on are added to the request, you can't commit unrelated changes to master until the request was merged. And you may need to force push master for requested changes.

**Comment by De-Seppe** [comment]
I made the necessary changes to the filename and the superfluous newline.

**Comment by Yamagi** [comment]
And merged. Thank you! :) I'll add `g_machinegun_norecoil` to the list of cheat protected cvars. That's mostly a placebo but at least we can claim that we tried to prevent abuse.

**Pull request merged**